### PR TITLE
Feature: add the Neo C# contract profile schema

### PR DESCRIPTION
**Pull request by Jim8y**
## Summary

- add a Draft 2020-12 schema for the Neo C# contract profile
- require state-specific diagnostics, implementation references, and evidence
- enforce SemVer 2.0 syntax for profile versions
- add a complete fixture and valid and invalid schema test matrices

## Motivation

Compiler support, analyzer enforcement, tests, and documentation currently have no shared compatibility contract. This schema defines stable capability IDs, support states, matching surfaces, diagnostics, semantic differences, and the evidence required before a capability enters the profile.

This pull request defines and validates the format only. It does not claim that the current C# capability inventory is complete.

## Validation

- 30 focused schema tests
- valid and invalid SemVer matrices, including prerelease and build metadata edge cases
- schema and fixture JSON parsing
- 230 analyzer tests with the seven audited fixes combined
- combined Release solution build
- combined full solution test run: 2,117 passed
- focused format and diff checks

Part of #1841.

### CODE REVIEW

File: profiles/neo-csharp-profile.schema.json [hunk 1]
```diff
@@ -0,0 +1,511 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
```
Comment (shargon) [inline profiles/neo-csharp-profile.schema.json:2]: How it's generated?

Comment (Jim8y) [reply profiles/neo-csharp-profile.schema.json:2]: It is not generated. This PR defines the versioned schema as the source of truth, and the tests validate the schema and its fixture. The complete capability inventory will be added separately and validated against it.

Comment (shargon) [reply profiles/neo-csharp-profile.schema.json:2]: And how it will be maintained, manually?

Comment (Jim8y) [reply profiles/neo-csharp-profile.schema.json:2]: The schema itself is a versioned format, so changes are reviewed manually. The capability inventory will not rely on manual synchronization alone: follow-up work will validate compiler handlers, analyzer diagnostics, tests, and generated documentation against it in CI.

**Review by shargon — DISMISSED** [review]

File: profiles/neo-csharp-profile.schema.json (hunk 1 above)
Comment (shargon) [reply profiles/neo-csharp-profile.schema.json:2]: It should be good have a readme about how modify it.

**Review by shargon — APPROVED** [review]

**Review by ajara87 — APPROVED** [review]
